src/Disks/IO/ReadBufferFromRemoteFSGather.cpp

```diff
@@ -75,14 +75,15 @@ size_t ReadBufferFromRemoteFSGather::readInto(char * data, size_t size, size_t o
 
     absolute_position = offset;
     bytes_to_ignore = ignore;
+    if (bytes_to_ignore)
+        assert(initialized());
 
     auto result = nextImpl();
-    bytes_to_ignore = 0;
 
     if (result)
-        return working_buffer.size();
+        return {working_buffer.size(), BufferBase::offset()};
```

Comment:
why we return offset? bytes_to_ignore already accounted?

Reply:
Here are inside a method `readInto(char * data, ...)` and in `ignore(bytes_to_ignore)` we read something and possibly there is pending data after `ignore` in current buffer. So in this case we return current buffer and offset -- to know that actual data starts from `offset` and not from `working_buffer.begin()`. This got a little complicated (adding `offset` as return parameter) because we cannot change where `data` in `readInto(char * data, ...)` points.